src/Encamina.Enmarcha.SemanticKernel.Connectors.Memory/Extensions/IServiceCollectionExtensions.cs

```diff
@@ -106,4 +106,71 @@ public static IServiceCollection AddSemanticTextMemory(this IServiceCollection s
                 .Build();
         });
     }
+
+    /// <summary>
+    /// Adds Azure AI Search as a named vector database for a memory store (<see cref="IMemoryStore"/>) to the <see cref="IServiceCollection"/> as a singleton service.
+    /// </summary>
+    /// <remarks>
+    /// This extension methods requires a <see cref="AzureAISearchOptions"/> to be already configured.
+    /// </remarks>
+    /// <param name="services"> The <see cref="IServiceCollection"/> to add services to.</param>
+    /// <param name="memoryProviderName">name of the memory provider.</param>
+    /// <returns>The <see cref="IServiceCollection"/> so that additional calls can be chained.</returns>
+    public static IServiceCollection AddAzureAISearchNamedMemoryStore(this IServiceCollection services, string memoryProviderName)
+    {
+        return services.AddKeyedSingleton<IMemoryStore>(memoryProviderName, (serviceProvider, k) =>
+        {
+            var optionsMonitor = serviceProvider.GetRequiredService<IOptionsMonitor<AzureAISearchOptions>>();
+
+            static AzureAISearchMemoryStore Builder(AzureAISearchOptions options)
+            {
+                return new AzureAISearchMemoryStore(options.Endpoint.AbsoluteUri, options.Key);
+            }
+
+            var debouncedBuilder = Debouncer.Debounce<AzureAISearchOptions>(options => Builder(options), 300);
+
+            var memory = Builder(optionsMonitor.CurrentValue);
+
+            optionsMonitor.OnChange(debouncedBuilder);
+
+            return memory;
+        });
+    }
+
+    /// <summary>
+    /// Adds Qdrant vector database as a named memory store (<see cref="IMemoryStore"/>) to the <see cref="IServiceCollection"/> as a singleton service.
+    /// </summary>
+    /// <remarks>
+    /// This extension methods requires a <see cref="QdrantOptions"/> to be already configured.
+    /// </remarks>
+    /// <param name="services"> The <see cref="IServiceCollection"/> to add services to.</param>
+    /// <param name="memoryProviderName">name of the memory provider.</param>
```

Comment:
Ídem 😉
Sería =>

`/// <param name="memoryProviderName">The name or key for the the memory provider.</param>`